File: Components/BlogML/Xml/BlogMLSerializer.vb
Imports System.IO
Imports System.Xml.Serialization
Imports System.Xml

Namespace BlogML.Xml
 Public Class BlogMLSerializer
  Private Shared ReadOnly syncRoot As New Object
  Private Shared m_serializer As XmlSerializer

  Public Shared ReadOnly Property Serializer As XmlSerializer
   Get
    SyncLock syncRoot
     If m_serializer Is Nothing Then
      m_serializer = New XmlSerializer(GetType(BlogMLBlog))
     End If
     Return m_serializer
    End SyncLock
   End Get
  End Property

  Public Shared ReadOnly Property Namespaces As XmlSerializerNamespaces
   Get
    Dim ns As New XmlSerializerNamespaces()
    ns.Add("dnn", "http://dnn-connect.org/blog/")
    Return ns
   End Get
  End Property

  Public Shared Function Deserialize(stream As Stream) As BlogMLBlog
   Return TryCast(Serializer.Deserialize(stream), BlogMLBlog)
  End Function

  Public Shared Function Deserialize(reader As TextReader) As BlogMLBlog
   Return TryCast(Serializer.Deserialize(reader), BlogMLBlog)
  End Function

  Public Shared Function Deserialize(reader As XmlReader) As BlogMLBlog
   Return TryCast(Serializer.Deserialize(reader), BlogMLBlog)
  End Function

  Public Shared Sub Serialize(stream As Stream, blog As BlogMLBlog)
   Serializer.Serialize(stream, blog, Namespaces)
  End Sub

  Public Shared Sub Serialize(writer As TextWriter, blog As BlogMLBlog)
   Serializer.Serialize(writer, blog, Namespaces)
  End Sub

  Public Shared Sub Serialize(writer As XmlWriter, blog As BlogMLBlog)
   Serializer.Serialize(writer, blog, Namespaces)
  End Sub
 End Class
End Namespace

File: Server/Core/BlogML/Xml/BlogMLSerializer.cs
using System.IO;
using System.Xml;
using System.Xml.Serialization;

namespace DotNetNuke.Modules.Blog.Core.BlogML.Xml
{
  public class BlogMLSerializer
  {
    private static readonly object syncRoot = new object();
    private static XmlSerializer m_serializer;

    public static XmlSerializer Serializer
    {
      get
      {
        lock (syncRoot)
        {
          if (m_serializer is null)
          {
            m_serializer = new XmlSerializer(typeof(BlogMLBlog));
          }
          return m_serializer;
        }
      }
    }

    public static XmlSerializerNamespaces Namespaces
    {
      get
      {
        var ns = new XmlSerializerNamespaces();
        ns.Add("dnn", "http://dnn-connect.org/blog/");
        return ns;
      }
    }

    public static BlogMLBlog Deserialize(Stream stream)
    {
      return Serializer.Deserialize(stream) as BlogMLBlog;
    }

    public static BlogMLBlog Deserialize(TextReader reader)
    {
      return Serializer.Deserialize(reader) as BlogMLBlog;
    }

    public static BlogMLBlog Deserialize(XmlReader reader)
    {
      return Serializer.Deserialize(reader) as BlogMLBlog;
    }

    public static void Serialize(Stream stream, BlogMLBlog blog)
    {
      Serializer.Serialize(stream, blog, Namespaces);
    }

    public static void Serialize(TextWriter writer, BlogMLBlog blog)
    {
      Serializer.Serialize(writer, blog, Namespaces);
    }

    public static void Serialize(XmlWriter writer, BlogMLBlog blog)
    {
      Serializer.Serialize(writer, blog, Namespaces);
    }
  }
}